pallets/parachain-staking/src/set.rs

```diff
@@ -124,6 +126,28 @@ impl<T: Ord> OrderedSet<T> {
 		self.0.binary_search(value)
 	}
 
+	// iterative search from left to right which is assumed to be sorted from
+	// greatest to least
+	pub fn linear_search(&self, value: &T) -> Result<usize, usize> {
```

Comment:
why not binary search if it's sorted?

Reply:
Have you checked the comparator? When you do binary search, it searches by amount first. However, we search for the account id half of the time, e.g. when changing the amount at stake or just checking for membership.

Thus, it does not work with binary search as we have 2 search parameters. We need to iterate over the whole list.